src/throttler/core.clj

```diff
@@ -166,8 +169,16 @@
        (fn [& args]
           ;; The approach is simple: pipe a bogus message through a
           ;; throttled channel before evaluating the original function.
-         (>!! in :eval-request)
-         (<!! out)
+         (let [[v _] (alts!! [[in :eval-request] (timeout max-queue-wait-timeout)])]
+           (when-not v
+             (throw (ex-info "Throttler in channel timed out"
+                             {:causes #{:in-ch-timed-out}
+                              :timeout max-queue-wait-timeout}))))
+         (let [[v _] (alts!! [out (timeout max-queue-wait-timeout)])]
+           (when-not v
+             (throw (ex-info "Throttler out channel timed out"
+                             {:causes #{:out-ch-timed-out}
```

Comment:
add some other keyword like :type :throttler-timeot or something so that we can use that in catch in try+ on our side